internal/tests/output.go

```diff
@@ -24,18 +24,56 @@ const (
 var (
 	// defaultFields is the set of fields that are ignored by default for any
 	// files by the given names.
-	defaultFields = map[string][]string{
+	defaultFields = map[string][][]strip.Step{
 		"apply.json": {
-			"0",
-			"*.@timestamp",
+
+			// The first message in the apply.json file is always the
+			// version information and other meta information.
+
+			strip.Field("0"),
+
+			// Remove the timestamp and hook.elapsed_seconds fields from the
+			// apply.json file. These will be different between runs and are not
+			// relevant to the test.
+
+			strip.Field("*.@timestamp"),
+			strip.Field("*.hook.elapsed_seconds"),
+
+			// Finally, remove the @message field from the apply_complete
+			// message. This contains information about how long the apply took
+			// and this is very flaky between runs.
+
+			{
+				{
+					Step: "*",
+					Filter: []strip.Filter{
+						{
+							Path:  []string{"type"},
+							Value: "apply_complete",
+						},
+					},
+				},
+				{
+					Step: "@message",
+				},
+			},
```

Comment:
I was stuck understanding this yesterday, but after looking at the test cases and using a debugger I understand that this instruction is written knowing that the contents of apply.json is a slice, so:

- the first step is selecting any elements from the slice where the `"type"` field ==  `"apply_complete"`
- then, for those elements only, it will proceed to remove the `"@message"` field

After getting my understanding of that, and how the code in strip.go works, I can see how this would address the flakiness in the test from last week.